src/server/player_sao.cpp

```diff
@@ -558,12 +558,35 @@ void PlayerSAO::setMaxSpeedOverride(const v3f &vel)
 
 bool PlayerSAO::checkMovementCheat()
 {
-	if (isAttached() || m_is_singleplayer ||
+	if (m_is_singleplayer ||
 			g_settings->getBool("disable_anticheat")) {
 		m_last_good_position = m_base_position;
 		return false;
 	}
 
+	ServerActiveObject *parentSAO = getParent();
+
+	if (parentSAO) {
+		UnitSAO *parent = dynamic_cast<UnitSAO *>(parentSAO);
+		if (! parent)
+			return false;
+		int parent_id;
+		std::string bone;
+		v3f attachment_pos, attachment_rot;
+		getAttachment(&parent_id, &bone, &attachment_pos, &attachment_rot);
+		v3f parent_pos = parent->getBasePosition();
+		f32 diff = (getBasePosition() - parent_pos).getLength() - (attachment_pos * 3).getLength();
```

Comment:
The following would be more efficient:
```suggestion
		f32 diff = m_base_position.getDistanceFromSQ(parent_pos) - attachment_pos.getLengthSQ() * 9.0f;
```